### PR TITLE
mary/evaluation_eancodeValidator

### PR DESCRIPTION
**Pull request by maryfranz**
Added code and tests for EAN Code validation

### CODE REVIEW

File: src/main/java/org/fundacionjala/coding/mary/EANValidator.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    private static int checksum(final String eanCode) {
+        int sum = 0;
+        int checksum = 0;
+        String[] digits = eanCode.split("");
+        int[] numbers = new int[digits.length];
+
+        for (int index = 0; index < digits.length; index++) {
+            numbers[index] = Integer.parseInt(digits[index]);
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/mary/EANValidator.java]: You don't need to save an array of integers

File: src/main/java/org/fundacionjala/coding/mary/EANValidator.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    private static final int MOD_10 = 10;
+    private static final int TOTAL = 10;
+    private static final int MULTIPLIED_BY_3 = 3;
+
+    private EANValidator() {
+    }
+
+    public static boolean validate(final String eanCode) {
+        double validateCode = (Double.parseDouble(eanCode)) % MOD_10;
+        return (validateCode == checksum(eanCode)) ? true : false;
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/mary/EANValidator.java]: the checksum should be compare with the last digit in the eanCode.
```java
return Character.getNumericValue(eanCode.charAt(eanCode.length() - 1)) == checksum(eanCode);
```

File: src/main/java/org/fundacionjala/coding/mary/EANValidator.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        int checksum = 0;
+        String[] digits = eanCode.split("");
+        int[] numbers = new int[digits.length];
+
+        for (int index = 0; index < digits.length; index++) {
+            numbers[index] = Integer.parseInt(digits[index]);
+        }
+
+        for (int i = 0; i < numbers.length - 1; i++) {
+            if ((i + 1) % 2 != 0) {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/mary/EANValidator.java]: You can use a ternary operation here.
```java
sum += (index + 1) % 2 != 0 ? number : number * MULTIPLIED_BY_3;
```

File: src/main/java/org/fundacionjala/coding/mary/EANValidator.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                sum = sum + numbers[i];
+            } else {
+                sum = sum + (numbers[i] * MULTIPLIED_BY_3);
+            }
+        }
+
+        if ((sum % MOD_10) != 0) {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/mary/EANValidator.java]: Same as above
```java
return sum % MOD_10 == 0 ? 0 : TOTAL - (sum % MOD_10);
```

**Review by carledriss — CHANGES_REQUESTED** [review]
After applying the changes
```java
    private static int checksum(final String eanCode) {
        int sum = 0;
        String[] digits = eanCode.split("");
        for (int index = 0; index < digits.length - 1; index++) {
            int number = Integer.parseInt(digits[index]);
            sum += (index + 1) % 2 != 0 ? number : number * MULTIPLIED_BY_3;
        }
        return sum % MOD_10 == 0 ? 0 : TOTAL - (sum % MOD_10);
    }
```

File: src/main/java/org/fundacionjala/coding/mary/EANValidator.java (hunk 2 above)
Comment (maryfranz) [reply src/main/java/org/fundacionjala/coding/mary/EANValidator.java]: Hi Carlos, 
Actually I'm getting the last digit of the code in the "validateCode" variable by applying the module 10 of the number in line 12. I guess the name of my variable was not good.
Thanks for the feedback, I will use that to update my code.

**Review by carledriss — APPROVED** [review]

**Pull request merged**